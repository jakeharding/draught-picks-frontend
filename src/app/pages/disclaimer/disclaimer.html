<!--
  Generated template for the DisclaimerPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>
  <draught-header></draught-header>
</ion-header>

<ion-content>
  <ion-row>
    <ion-col class="ion-text-center"
             size="12" sizeSm="10" offsetSm="1" sizeMd="8" offsetMd="2" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4">
      <h1>Disclaimer</h1>
      <h4>Please drink responsibly.</h4>
      <div class="justify">
        <p margin>Draught Picks is not responsible for any over consumption of alcohol.
          By using this application you acknowledge your responsibility for your actions and exclude Draught Picks
          and the creators of Draught Picks from any liability without regard to your mental state.
          By using this application, you agree you will not pursue any litigation against Draught Picks and its creators
          at any time for any reason.
        </p>
        <p margin>You also acknowledge that you are over 21 by using this application. Draught Picks is not responsible for under age drinking.</p>
      </div>
      <a tid="disclaimerVerified" routerLink="/register" routerDirection="back">I have read the disclaimer</a>
    </ion-col>
  </ion-row>
</ion-content>
