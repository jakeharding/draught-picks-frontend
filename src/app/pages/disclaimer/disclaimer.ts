import { Component } from '@angular/core';
import { BasePage } from '../BasePage';

/**
 * Generated class for the DisclaimerPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@Component({
  selector: 'page-disclaimer',
  templateUrl: 'disclaimer.html',
  styleUrls: ['./disclaimer.scss']
})
export class DisclaimerPage extends BasePage {


  /**
   * DisclaimerPage constructor
   * empty but required constructor for the DisclaimerPage Class, does nothing
   */
  constructor() {
    super('disclaimer');
  }
}
