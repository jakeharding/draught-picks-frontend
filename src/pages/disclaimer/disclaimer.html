<!--
  Generated template for the DisclaimerPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>

  <ion-navbar color="primary">
    <ion-title text-center>Draught Picks</ion-title>
  </ion-navbar>

</ion-header>


<ion-content text-center>
  <h1>Disclaimer</h1>
  <p margin>Please drink responsibly. Draught Picks is not responsible for any over consumption of alchohol. By reading this Disclaimer,
  you acknowledge that you are responsible for the amount of alcohol consumed.
  </p>
  <p margin>You also acknowledge that you are over 21 by using this application. Draught Picks is not responsible for under age drinking.</p>

  <button ion-button navPop>I have read the disclaimer</button>
</ion-content>
