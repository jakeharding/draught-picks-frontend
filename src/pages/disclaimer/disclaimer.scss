page-disclaimer {
  .back-button.show-back-button{
    display: none;
  }
}
