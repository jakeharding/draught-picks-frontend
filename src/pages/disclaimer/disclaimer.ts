import { Component } from '@angular/core';
import { IonicPage, NavController, NavParams } from 'ionic-angular';
import { BasePage } from "../BasePage";

/**
 * Generated class for the DisclaimerPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@IonicPage()
@Component({
  selector: 'page-disclaimer',
  templateUrl: 'disclaimer.html',
})
export class DisclaimerPage extends BasePage {


  /**
   * DisclaimerPage constructor
   * empty but required constructor for the DisclaimerPage Class, does nothing
   * */
  constructor(public navCtrl: NavController, public navParams: NavParams ) {
    super('disclaimer');
  }
}
