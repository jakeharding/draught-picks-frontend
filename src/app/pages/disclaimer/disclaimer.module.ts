import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { DisclaimerPage } from './disclaimer';
// import { ComponentsModule } from '../../components/components.module';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [
    DisclaimerPage,
  ],
  imports: [
    IonicModule,
    RouterModule.forChild([{path: 'disclaimer', component: DisclaimerPage}])
    // ComponentsModule,
  ],
})
export class DisclaimerPageModule {}
