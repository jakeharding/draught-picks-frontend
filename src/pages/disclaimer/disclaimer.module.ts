import { NgModule } from '@angular/core';
import { IonicPageModule } from 'ionic-angular';
import { DisclaimerPage } from './disclaimer';
import { ComponentsModule } from "../../components/components.module";

@NgModule({
  declarations: [
    DisclaimerPage,
  ],
  imports: [
    IonicPageModule.forChild(DisclaimerPage),
    ComponentsModule,
  ],
})
export class DisclaimerPageModule {}
