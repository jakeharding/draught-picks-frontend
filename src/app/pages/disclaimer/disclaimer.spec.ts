import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { DisclaimerPage } from './disclaimer';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';

/**
 * disclaimer.spec.ts
 *
 * Created by jake
 * Created on 11/18/18
 *
 * Test DisclaimerPage.
 */

describe('DisclaimerPage', () => {
  let fixture: ComponentFixture<DisclaimerPage>;
  let instance: DisclaimerPage;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [DisclaimerPage],
      providers: []
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DisclaimerPage);
    instance = fixture.debugElement.componentInstance;
  });

  it('should initialize the DisclaimerPage', () => {
    expect(instance).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
