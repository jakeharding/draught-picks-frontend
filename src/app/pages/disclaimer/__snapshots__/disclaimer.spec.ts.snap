// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisclaimerPage should initialize the DisclaimerPage 1`] = `
<page-disclaimer>
  <draught-header /><ion-content>
    <ion-row>
      <ion-col
        class="below-header"
        col-12=""
        col-lg-6=""
        col-md-8=""
        col-sm-10=""
        col-xl-4=""
        offset-lg-3=""
        offset-md-2=""
        offset-sm-1=""
        offset-xl-4=""
        text-center=""
      >
        <h1>
          Disclaimer
        </h1>
        <h4>
          Please drink responsibly.
        </h4>
        <div
          class="justify"
        >
          <p
            margin=""
          >
            Draught Picks is not responsible for any over consumption of alchohol. By using this application you acknowledge your responsibility for your actions and exclude Draught Picks and the creators of Draught Picks from any liability without regard to your mental state. By using this application, you agree you will not pursue any litigation against Draught Picks and its creators at any time for any reason. 
          </p>
          <p
            margin=""
          >
            You also acknowledge that you are over 21 by using this application. Draught Picks is not responsible for under age drinking.
          </p>
        </div>
        <a
          navpop=""
          tid="disclaimerVerified"
        >
          I have read the disclaimer
        </a>
      </ion-col>
    </ion-row>
  </ion-content>
</page-disclaimer>
`;
